### ISSUE TITLE
[BUG] Error: default auth: metadata-service: token response: Get "http://127.0.0.1:53405/49c4dd50-29b4-47f6-a345-e41fbad5d65f": dial tcp 127.0.0.1:53405: connectex: No connection could be made because the target machine actively refused it

### ISSUE THREAD

**Issue by SrihasaAkepati** [databricks/databricks-vscode#980]
**Describe the bug**

Visual Studio code extension for databricks exhibits a behaviour where databricks cli commands run from the terminal intermittently throws an error "_default auth: metadata-service: token response: Get "http://127.0.0.1:53405/49c4dd50-29b4-47f6-a345-e41fbad5d65f": dial tcp 127.0.0.1:53405: connectex: No connection could be made because the target machine actively refused it_". This is not a consistent but rather a random behaviour. Commands run from terminal using a random directory works sometimes but fails from other directories. 

**To Reproduce**

-windows 11
-visual studio code
- Install the Databricks CLI and extension accordingly to the documentation:
databricks cli- https://learn.microsoft.com/en-us/azure/databricks/dev-tools/cli/install#--source-installation-for-linux-macos-and-windows
databricks extension - 
https://learn.microsoft.com/en-us/azure/databricks/dev-tools/vscode-ext/install
we installed the databricks extension (1.2.4)
- Open a new folder in visual studio code, configure databricks profile from this directory and run dbcli commands. ex : databricks clusters list 

**Screenshots**

Databricks configuration file 

<img width="925" alt="image" src="https://github.com/databricks/databricks-vscode/assets/93312070/5a93bb05-d65b-4c78-99e7-f4c7c90bb1ed">

<img width="1555" alt="image" src="https://github.com/databricks/databricks-vscode/assets/93312070/13b57728-fc2e-4073-bab0-156cc37aa8b8">

After replacing metadata_service_url in databricks cfg file with metadata_service_url from .databricks.env file of a directory from visual studio code we're able to run the commands

<img width="1636" alt="image" src="https://github.com/databricks/databricks-vscode/assets/93312070/f28b249f-d461-4d0b-85af-cc8c7950543b">

One additional test we did is to kill a session which is actively listening on a working port. 

<img width="865" alt="image" src="https://github.com/databricks/databricks-vscode/assets/93312070/e4f5baed-fb7b-480b-a367-6f21dfd81c48">

<img width="1633" alt="image" src="https://github.com/databricks/databricks-vscode/assets/93312070/dd9e5b36-33b9-47a9-8ec8-bcbfdf988af4">

The port number 53405 was actually accepting connections before we killed it and after killing the session associated with this port we see the same connection refused error. 

In summary -

When we have two folders configured with databricks extension (which generates .databricks.env file in each folder with unique port number/unique id for metadata_service_url), dbcli commands work from one folder but fail from another folder.

**System information:**

1. Paste the output ot the `Help: About` command (CMD-Shift-P). Windows 11
2. Databricks Extension Version

**Databricks Extension Logs**
Please attach the [databricks extension logs](https://docs.databricks.com/dev-tools/vscode-ext.html#send-usage-logs-to-databricks)

**Additional context**

**Comment by mahhafez** [comment]
Dear @ilia-db Do we have an update on this issue it is blocking our cutsomer?

**Comment by kartikgupta-db** [comment]
Hi. The root cause of the issue is that environment variables are not updated properly by VS Code in the terminal. This issue is also reproduceable consistently. We are working on a fix. We should have something in the post holiday release.
